src/main/java/seedu/address/logic/parser/Autocompleter.java

```diff
@@ -0,0 +1,142 @@
+        // with what the user has currently typed
+
+        return getCommandList().stream().filter(command -> command.startsWith(input)).collect(
+                Collectors.toList()).size() > 0;
+
+    }
+
+    /**
+     * Filters from list of available autocompletions and retrieves the first matching completion
+     */
+    public static String getAutocompletion(String input) {
+        return getAutocompletionStrings().stream().filter(
+            completion -> completion.startsWith(input)).collect(Collectors.toList()).get(0);
+    }
+
+    private static List<String> getCommandList() {
+        return getCommandField("COMMAND_WORD");
+
+    }
+
+    private static List<String> getAutocompletionStrings() {
+        return getCommandField("AUTOCOMPLETE_TEXT");
+
+    }
+
+    /**
+     * Get a property/field for a particular command class.
+     */
+    private static List<String> getCommandField(String field) {
+        return getCommandClasses().stream().map(command -> {
+            try {
+                // Get all fields in the given list of strings
+                return (String) command.getField(field).get(null);
+            } catch (IllegalAccessException e) {
+                return "";
+
+            } catch (NoSuchFieldException e) {
+                return "";
+            }
+        }).collect(Collectors.toList());
+
+    }
+
+    /**
+     * Creates a list of all existing command classes and returns it to the user.
+     */
+    private static List<Class<? extends Command>> getCommandClasses() {
+        //        File directory = new File(COMMAND_DIRECTORY);
+        //        File[] fList = directory.listFiles();
+        List<Class<? extends Command>> commandClasses = new ArrayList<>();
+        commandClasses.add(ChangeDeckCommand.class);
+        commandClasses.add(ClassifyCommand.class);
+        commandClasses.add(ClearCommand.class);
+        commandClasses.add(DeleteCardCommand.class);
+        commandClasses.add(DeleteDeckCommand.class);
+        commandClasses.add(EditCardCommand.class);
+        commandClasses.add(EditDeckCommand.class);
+        commandClasses.add(EndReviewCommand.class);
+        commandClasses.add(ExitCommand.class);
+        commandClasses.add(ExportDeckCommand.class);
+        commandClasses.add(FindCommand.class);
+        commandClasses.add(FlipCardCommand.class);
+        commandClasses.add(HelpCommand.class);
+        commandClasses.add(HistoryCommand.class);
+        commandClasses.add(ImportDeckCommand.class);
+        commandClasses.add(ListCommand.class);
+        commandClasses.add(NewCardCommand.class);
+        commandClasses.add(NewDeckCommand.class);
+        commandClasses.add(PreviousCardCommand.class);
+        commandClasses.add(RedoCommand.class);
+        commandClasses.add(ReviewCommand.class);
+        commandClasses.add(SortCommand.class);
+        commandClasses.add(UndoCommand.class);
+
+
+        //        for (File file : fList) {
```

Comment:
What is this section?

Reply:
I'm still considering whether to search the file directory instead of manually adding all the commands. Thus, I've decided to comment out that chunk of code first until I make a decision on this tomorrow.

Reply:
Should I decide not to use it, it will be removed in the follow up pull request